aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -392,6 +410,22 @@ impl AptosVM {
         self.move_vm.env.clone()
     }
 
+    fn should_perform_async_runtime_checks(&self, txn: &SignedTransaction) -> bool {
+        self.async_runtime_checks_enabled
+            && match txn.payload().executable_ref() {
+                Ok(TransactionExecutableRef::Script(_)) => {
+                    // For now, always delay checks for scripts.
+                    true
+                },
+                Ok(TransactionExecutableRef::EntryFunction(f)) => {
+                    // If entry function is defined at special address - it is part of trusted code
+                    // and so no need to delay any checks (as there are none).
+                    !f.module().address().is_special()
+                },
+                Ok(TransactionExecutableRef::Empty) | Err(_) => false,
```

Comment:
Empty can refer to a multisig payload on-chain I think, false means the check is done in-place?